### PR TITLE
jdTextEdit: Add Recipe

### PR DESCRIPTION
**Pull request by JakobDev**
jdTextEdit is a Text Editor made by me.

The program depends on editorconig, so we need to wait until #5660 is merged, so I can add it to the requirements.

I also don't know how to add a Icon to the Deskbar.

### CODE REVIEW

File: app-editors/jdtextedit/jdtextedit-8.2.recipe [hunk 1]
```diff
@@ -0,0 +1,58 @@
+SUMMARY="A advanced text editor"
+DESCRIPTION="A advanced Text Editor which has usefull features: \
```
Comment (diversys) [inline app-editors/jdtextedit/jdtextedit-8.2.recipe:2]: Drop `\` here and below.

File: app-editors/jdtextedit/jdtextedit-8.2.recipe [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+{
+	"enableUpdater": false,
+	"aboutMessage": "jdTextEdit was installed from HaikuDepot"
+}
+EOF
+
+	mkdir -p $appsDir
+	mv $prefix/bin/jdTextEdit $appsDir/jdTextEdit
+	settype -t application/x-vnd.Be-elfexecutable $appsDir/jdTextEdit
+	addAppDeskbarSymlink $appsDir/jdTextEdit "jdTextEdit"
```
Comment (diversys) [inline app-editors/jdtextedit/jdtextedit-8.2.recipe:57]: `addAppDeskbarSymlink $appsDir/jdTextEdit` should be enough.

**Comment by JakobDev** [comment]
@diversys Your requested changes are now done. I also added editorconfig_core_py_python38 to the requirements. Now the only problem that I have to solve is how to set a Icon.

**Comment by korli** [comment]
> @diversys Your requested changes are now done. I also added editorconfig_core_py_python38 to the requirements. Now the only problem that I have to solve is how to set a Icon.

Did you look at the picard recipe: https://github.com/haikuports/haikuports/blob/f1552678e4fbe890c8d47bc87548a03ddd79623f/media-sound/picard/additional-files/picard.rdef.in#L52

**Comment by JakobDev** [comment]
@korli How can I convert a png to a icon array?

**Comment by korli** [comment]
> @korli How can I convert a png to a icon array?
Here are some links:
https://discuss.haiku-os.org/t/changing-an-icon-howto/3070/7

**Comment by humdingerb** [comment]
> How can I convert a png to a icon array?

WonderBrush can export as source code rdef. From there 'rc' can convert to a resource.

**Comment by pulkomandy** [comment]
The wonderbrush export will not get you a vector icon, and I don't think it will even get you an old BeOS-style bitmap icon. So you can put that in a resource, but it will not show in Tracker.

Do you have an SVG for the icon? If so, you can convert that using Icon-O-Matic to the proper format. If you only have a bitmap you'll need to create a vector version somehow.

**Comment by humdingerb** [comment]
I thought the question was how to get existing PNGs into a resource to be used in-app, e.g. tool bar icons. Vector would be prefered, ofc.

**Comment by JakobDev** [comment]
Unfortunately the [Logo](https://gitlab.com/JakobDev/jdTextEdit/-/blob/master/jdTextEdit/Logo.png) is only available as PNG and as GIMP File. In this case I think it's the best to keep the Entry in the Application Menu without a Icon., because I'm not good in creating Icons. That means that the recipe is ready to merge.

File: app-editors/jdtextedit/jdtextedit-8.2.recipe [hunk 3]
```diff
@@ -0,0 +1,55 @@
+SUMMARY="A advanced text editor"
+DESCRIPTION="A advanced Text Editor which has usefull features:
```
Comment (humdingerb) [inline app-editors/jdtextedit/jdtextedit-8.2.recipe:2]:
```suggestion
DESCRIPTION="An advanced text editor, written in Python, with useful features:
```

File: app-editors/jdtextedit/jdtextedit-8.2.recipe [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+* Sytax Highlighting
```
Comment (humdingerb) [inline app-editors/jdtextedit/jdtextedit-8.2.recipe:3]:
```suggestion
* Sytax highlighting
```

**Comment by humdingerb** [comment]
This should work as a resource file to be added with 'addResourcesToBinaries' in the recipe:

```
resource app_signature "application/x-vnd.jdtextedit";

resource app_flags B_MULTIPLE_LAUNCH;

resource app_version {
	major  = 8,
	middle = 2,
	minor  = 0,

	variety = B_APPV_FINAL,
	internal = 0,

	short_info = "An advanced text editor",
	long_info = "An advanced text editor"
};

resource large_icon array {
	$"FFFFFFFFFFFFFFFFFFFFFFFFDDDDD5D4D4D4D4D4FFFFFFFFFFFFFFFFFFFFFFFF"
	$"FFFFFFFFFFFFFFFFFFD5DDDDDDDDDDDDDDD4D4D4D4D4D4FFFFFFFFFFFFFFFFFF"
	$"FFFFFFFFFFFFFFD4D4D4D5DD27272727272726DDD4D4D4D5DDFFFFFFFFFFFFFF"
	$"FFFFFFFFFFFFD3D4D4DC2626272727272727262625DCD4D4DDDDFFFFFFFFFFFF"
	$"FFFFFFFFFFB3B3D3D4252526262626262626262625D4D3D3D4DDDDFFFFFFFFFF"
	$"FFFFFFFFD3B3D3D3D425262626DDDCDCDCDD26262625D3D3D3D4DD27FFFFFFFF"
	$"FFFFFFD4D4D3D3D3DC26DDDCD4D4D4D4D4D4D4D4D52626D4D3D3D4DD04FFFFFF"
	$"FFFFD4D4D3D3D3D426DCD4D4D4D4D4D4D4D4D487AEDC2626D4D3D3D42828FFFF"
	$"FFFFDDD4D3D3D425DCD4D4D3D3D3D3D3D3ADADAD87B4D4DDD5AED3D32728FFFF"
	$"FFDDDDD4D3D3DDDCD4D4D3D3D3D3D3D3D3878D87ADADD4D4AEAED3D3DC2829FF"
	$"FF27DDD3D3D4DCD4D4D3D3D3D3D3B3ADAD87878DADD4B4AED5DDDCD3D42729FF"
	$"FF28DCD3D426D4D4D3D3D3D3B3B3878DAD8787ADD3AEAED4DD2626D4D42629FF"
	$"2804DCD3DCDCD4D4D3D3D3B3ADAD8717AD878DD3AEAED4D4DC252625DC252929"
	$"2928DCD425DCD4D3D3D3D3ADAD8787878787ADAEADD4D4D4D4DC2626DCDC2829"
	$"2928DCDC26DCD4D3D3D3B3ADAD87878787ADAED4D4D4D4D4D4DC2626DCD42829"
	$"2928DCDC26DCD4D3D3D3B3AD87878787AD87ADD4D4D3D3D4D4DC2626D4D42729"
	$"2927D4DC26DCD4D3D3ADAD8787878DADADAE87D3D3D3D3D4D4DC2626D4D32729"
	$"2927D4D426DCD4D387AD87878DADADAEB4D4D3D3D3D3D3D4D4DC2626D3D32729"
	$"2928D3D426DCD4AD87878787ADADAEB4D4D3D3D3D3D3D4D4D4DD2625D3D32829"
	$"2828D4D3DC25D4D4D387ADADB3B4D4D4D3D3D3D3D3D3D4D4DC2626DCD3D42929"
	$"FF28DCD3D426D4D48DD3ADD3B4D4D4D3D3D3D3D3D3D4D4D4DC2626D3D3DD29FF"
	$"FF27DDD3D325D5AE87B3D3AEAED4D3D3D3D3D3D3D4D4D4DC262625D3D32702FF"
	$"FFDDDDD4D3D426D4AED4AE87D4D4D3D3D3D3D3D4D4D4DC252626D4D3DC2829FF"
	$"FFFFDDD4D3D3DC26D4AE11D4D4D4D4D4D4D4D4D4D4DC252626DCD4D42728FFFF"
	$"FFFFD4D4D4D4D3D4AE0FD5DCD4D4D4D4D4D4D4DCDC26262625D4DCDD28FFFFFF"
	$"FFFFFFD4D4D4D4D4AED52625DCDCDCDCDCDCDC2626262625DCDCDDDE04FFFFFF"
	$"FFFFFFFFD3D3D4D4DCDC262626262626262626262626252425DDDD27FFFFFFFF"
	$"FFFFFFFFFFB3B3D4DC252425262626262626262625252526DDDDDDFFFFFFFFFF"
	$"FFFFFFFFFFFFD3D4D4DC262525252525252526262626DDD5DDDDFFFFFFFFFFFF"
	$"FFFFFFFFFFFFFFD4D4D4D5DD2626262627272626DDD4D4D5DDFFFFFFFFFFFFFF"
	$"FFFFFFFFFFFFFFFFFFD5DDDDDDDDDDDDD5D4D4D4D4D4D4FFFFFFFFFFFFFFFFFF"
	$"FFFFFFFFFFFFFFFFFFFFFFFFDDDDD5D4D4D4D4D4FFFFFFFFFFFFFFFFFFFFFFFF"
};

resource mini_icon array {
	$"FFFFFFFFFFDDDDDDD4D4D4FFFFFFFFFF"
	$"FFFFFFD4D42627272726DCD4DDFFFFFF"
	$"FFFFB3D3DC262626262625D3D4DEFFFF"
	$"FFD4D3D325D4D4D4D4D4D425D3D428FF"
	$"FFD4D3DDD4D3D3D3AD8787D4AED327FF"
	$"28D4D4D4D3D3D3878787D4AEDDDCDC29"
	$"28D4DDD4D3B3AD8787ADAED4DC26DC29"
	$"29DCDCD4D3AD8787ADAED4D4DC26D429"
	$"28D4DCD4AD8787ADB4B3D3D4DC26D329"
	$"28D326D48D8DADB4D3D3D3D4DC26D329"
	$"28D4DCD4ADD3D4D3D3D3D4D426D4DC29"
	$"FFDDD3DDD4AED4D3D3D4D42626D428FF"
	$"FFD4D4D4AEDDDCD4DCDC2626DCDD28FF"
	$"FFFFB3D4DC25262626262525DD27FFFF"
	$"FFFFFFD4D42626262626DDD5DDFFFFFF"
	$"FFFFFFFFFFDDDDD5D4D4D4FFFFFFFFFF"
};

```

**Comment by JakobDev** [comment]
addResourcesToBinaries doesn't work with Python, but I find another way in the recipe of [QuteBrowser](https://github.com/haikuports/haikuports/blob/master/www-client/qutebrowser/qutebrowser-1.13.1.recipe).

File: app-editors/jdtextedit/jdtextedit-8.2.recipe (hunk 4 above)
Comment (korli) [reply app-editors/jdtextedit/jdtextedit-8.2.recipe]: "Syntax"

**Comment by Begasus** [comment]
This failed to build on 32-bit:
```
resolving build dependencies failed: requires "qscintilla_python38" of package "jdtextedit-8.2" could not be resolved
```
https://build.haiku-os.org/buildmaster/master/x86_gcc2/?buildrunDir=1184&viewMode=expanded

**Pull request merged**